docs/highlighters/regexp.md

```diff
@@ -28,3 +28,19 @@ in [the `zshmisc(1)` manual page][zshmisc-Conditional-Expressions]
 [zshzle-Character-Highlighting]: http://zsh.sourceforge.net/Doc/Release/Zsh-Line-Editor.html#Character-Highlighting
 [perlretut]: http://perldoc.perl.org/perlretut.html
 [zshmisc-Conditional-Expressions]: http://zsh.sourceforge.net/Doc/Release/Conditional-Expressions.html#Conditional-Expressions
+
+### Cautions to BSD-based platform users
+
+In BSD-based platform such as macOS, the regex metacharacters with leading backslash (`\`)
+such as `\b`, `\w`, etc. are [not supported](https://stackoverflow.com/a/12696899).
```

Comment:
We needn't document the precise differences between platforms here; that would be liable to get out of date.  Simply stating that there are platform differences and pointing to the syntax documentation would suffice, IMO.  WDYT?